### PR TITLE
[Fix] Error : Unknown named parameter $var

### PR DESCRIPTION
**Pull request by lloricode**
when using array as same in v2 will return:
Error : Unknown named parameter $var

Edit:
see https://github.com/lloricode/paymaya-sdk-php/runs/2564106914?check_suite_focus=true

### CODE REVIEW

**Comment by brendt** [comment]
Can you provide a test and an example?

**Comment by brendt** [comment]
(pressed the wrong button, didn't mean to close)

**Comment by lloricode** [comment]
> Can you provide a test and an example?

yes i will :)

File: tests/ArrayCaster/ArrayV2CasterTest.php [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+use Spatie\DataTransferObject\DataTransferObject;
+use Spatie\DataTransferObject\Tests\TestCase;
+
+class ArrayV2CasterTest extends TestCase
+{
+    /** @test */
+    public function version_2_will_casted()
+    {
+        $bar = new ParentDto(
```
Comment (lloricode) [inline tests/ArrayCaster/ArrayV2CasterTest.php:13]: Hi @brendt,

this is our actual specific error when upgrading to v3.

Thank you

Comment (lloricode) [reply tests/ArrayCaster/ArrayV2CasterTest.php:13]: on our project, its too difficult to refactor the DTO, since many using on this object

**Comment by brendt** [comment]
If I understand you correct, you want to have interoperability between v2 and v3 objects?

**Comment by lloricode** [comment]
> If I understand you correct, you want to have interoperability between v2 and v3 objects?

yes, we want v2 will completely work with v3 without any changes

**Comment by brendt** [comment]
That's unfortunately not going to happen. We don't want to maintain interop between major versions.

**Pull request closed without merging**